hardware/VirtualThermostat.cpp

```diff
@@ -0,0 +1,740 @@
+#include "stdafx.h"
+#include "../main/Helper.h"
+#include "../main/Logger.h"
+#include "../main/SQLHelper.h"
+#include "../main/RFXtrx.h"
+
+#include "VirtualThermostat.h"
+
+#include "../main/localtime_r.h"
+#include "../main/mainworker.h"
+#include "../notifications/NotificationHelper.h"
+
+#include <map>
+
+std::string GetDeviceValue(const char * FieldName, const char *Idx)
+{
+	TSqlQueryResult result = m_sql.safe_query("SELECT %s from DeviceStatus WHERE (ID == %s )", FieldName, Idx);
```

Comment:
we use the 'auto' for these results in our code these days

auto result = m_sql.sa...

Reply:
ok ! done ! new for me .